src/main/java/org/jabref/logic/importer/fetcher/MergingIdBasedFetcher.java

```diff
@@ -0,0 +1,121 @@
+
+    private Optional<FetcherResult> fetchEntry(IdBasedFetcher fetcher, Field field,
+                                               String identifier, BibEntry originalEntry) {
+        try {
+            LOGGER.debug("Attempting to fetch entry using {} fetcher", field);
+            logEntryDetails(originalEntry, field, identifier);
+
+            return fetcher.performSearchById(identifier)
+                          .map(fetchedEntry -> {
+                              BibEntry mergedEntry = (BibEntry) originalEntry.clone();
+                              boolean hasChanges = mergeBibEntry(mergedEntry, fetchedEntry);
+                              return new FetcherResult(originalEntry, mergedEntry, hasChanges);
+                          });
+        } catch (Exception exception) {
+            LOGGER.error("Error fetching entry with {} identifier: {}",
+                    field, identifier, exception);
+            return Optional.empty();
+        }
+    }
+
+    private boolean mergeBibEntry(BibEntry target, BibEntry source) {
+        boolean hasChanges = false;
+        for (Field field : source.getFields()) {
+            Optional<String> sourceValue = source.getField(field);
+            Optional<String> targetValue = target.getField(field);
+            if (sourceValue.isPresent() && !sourceValue.equals(targetValue)) {
+                target.setField(field, sourceValue.get());
+                hasChanges = true;
+            }
+        }
+        return hasChanges;
+    }
+
+    private Optional<IdBasedFetcher> getFetcherForField(Field field) {
+        return switch (field) {
+            case StandardField.DOI -> Optional.of(new DoiFetcher(importFormatPreferences));
+            case StandardField.ISBN -> Optional.of(new IsbnFetcher(importFormatPreferences));
+            case StandardField.EPRINT -> Optional.of(new IacrEprintFetcher(importFormatPreferences));
+            default -> Optional.empty();
+        };
+    }
+
+    private void logEntryIdentifiers(BibEntry entry) {
+        List<String> availableIds = Stream.of(StandardField.DOI, StandardField.ISBN, StandardField.EPRINT)
+                                          .flatMap(field -> entry.getField(field)
+                                                                 .map(value -> field +
+                                                                         ": " +
+                                                                         value)
+                                                                 .stream())
+                                          .collect(Collectors.toList());
+
+        String citationKey = entry.getCitationKey()
+                                  .map(key -> "citation key: " + key)
+                                  .orElse("no citation key");
+
+        LOGGER.debug("Processing entry with {} and identifiers: {}",
+                citationKey,
+                availableIds.isEmpty() ? "none" : String.join(", ", availableIds));
```

Comment:
I think, this method can just be removed. `LOGGER.debug("Entry {}", entry)` is really enough. 

I know that for your development, this was needed. But when developing longer with JabRef, one can read BibTeX output of `org.jabref.model.entry.CanonicalBibEntry#getCanonicalRepresentation`.